assert/assertions.go

```diff
@@ -1400,6 +1421,25 @@ func DirExists(t TestingT, path string, msgAndArgs ...interface{}) bool {
 	return true
 }
 
+// NoDirExists checks whether a directory does not exist in the given path.
+// It fails if the path points to an existing _directory_ only.
+func NoDirExists(t TestingT, path string, msgAndArgs ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	info, err := os.Lstat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return true
+		}
+		return true
+	}
+	if !info.IsDir() {
+		return true
+	}
```

Comment:
Would a symlink to a directory fail this check? If so, we should probably try to make the file and dir versions consistent in how picky they are.

Reply:
Agreed. Will check but it might be a breaking change if File and Dir are not consistent with regards to symlinks.